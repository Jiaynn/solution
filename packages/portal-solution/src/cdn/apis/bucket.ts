import { injectable } from 'qn-fe-core/di'

import { prefix } from 'cdn/constants/api'

import CommonClient from './clients/common'

export interface IBucketSimplified {
  name: string,
  zone: string,
  private: number,
  protected: number,
  last_operation: string,
  last_operation_extra: string,
  last_operation_at: string,
  fop_access_white_list: string[],
  files: number,
  storage: number,
  line_files: number,
  line_storage: number,
  right: string,
  share: boolean
}

// 由于 bucket 未接入 iam，因此暂时提供 /api/fusion 开头的接口用于获取 bucket 相关信息

@injectable()
export default class BucketApis {
  constructor(private client: CommonClient) {}

  getBucketsSimplified() {
    return this.client.get<IBucketSimplified[]>(`${prefix}/buckets-simplified`)
  }
}
