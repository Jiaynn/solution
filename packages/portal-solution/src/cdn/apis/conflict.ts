import { injectable } from 'qn-fe-core/di'

import { prefix } from 'cdn/constants/api'

import CommonClient from './clients/common'
import DomainProxyClient from './clients/domain-proxy'

export interface IGetVerifyFileResp {
  rtn: string
  msg: string
  findDomain: string
}

export interface IVerifyParam {
  conflictDomain: string
  targetprod?: string
}

@injectable()
export default class ConflictApis {
  constructor(
    private commonClient: CommonClient,
    private domainProxyClient: DomainProxyClient
  ) {}

  getVerifyFile(domain: string) {
    return this.domainProxyClient.get<IGetVerifyFileResp>(`/findback/downloadverifyfile/${domain}`)
  }

  verifyDomain(verifyParam: IVerifyParam) {
    // 产品线默认为 fusion
    return this.commonClient.post<unknown>(`${prefix}/verify`, {
      ...verifyParam,
      targetprod: verifyParam.targetprod ?? 'fusion'
    })
  }
}
