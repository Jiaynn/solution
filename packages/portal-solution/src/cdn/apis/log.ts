/**
 * @file log apis
 * @author gaoupon <gaopeng01@qiniu.com>
 */

import { injectable } from 'qn-fe-core/di'

import DefyLogClient from './clients/defy-log'

export interface IFilterOptions {
  domains: string[]
  start: number
  end: number
  marker: string
  limit: number
}

export interface ILog {
  name: string
  size: number
  time: number
  url: string
}

export interface ITask {
  domain: string
  start: number
  end: number
  finish: number
  status: string
  logs: ILog[]
}

@injectable()
export default class LogApis {
  constructor(
    private client: DefyLogClient
  ) { }

  getFusionLogs(param: IFilterOptions): Promise<ITask[]> {
    return this.client.post<{data: ITask[]}>('/v2/log/tasks', param).then(
      res => (res.data || [])
    )
  }
}
