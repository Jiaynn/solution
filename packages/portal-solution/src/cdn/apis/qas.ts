/*
 * @file 质量保障服务相关接口
 * @author linchen <linchen@qiniu.com>
 */

import { injectable } from 'qn-fe-core/di'

import { prefix } from 'cdn/constants/api'
import { PrimeLevel, PrimeStatus } from 'cdn/constants/qas'

import CommonClient from './clients/common'

export interface IPrimeItem {
  level: PrimeLevel // level
  sla: string       // sla 等级
  ratio: number     // 赔付比率
  price: number     // 相对于原单价的比例
}

export interface IQasPrimeStatus {
  level: PrimeLevel // 等级
  state: PrimeStatus  // 状态
}

export interface IQasInfo {
  levelstates: IQasPrimeStatus[]
}

@injectable()
export default class QasApis {
  constructor(private client: CommonClient) {}

  getQasInfo() {
    return this.client.get<IQasInfo>(`${prefix}/insurance/get`)
  }

  cancelPrime(apply: PrimeLevel) {
    return this.client.post<void>(`${prefix}/insurance/cancel`, { apply })
  }

  openPrime(apply: PrimeLevel) {
    return this.client.post<void>(`${prefix}/insurance/set`, { apply })
  }
}
