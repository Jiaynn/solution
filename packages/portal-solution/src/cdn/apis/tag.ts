/**
 * @file tags apis
 * @author linchen <gakiclin@gmail.com>
 */

import { injectable } from 'qn-fe-core/di'

import DomainProxyClient from './clients/domain-proxy'

export interface DomainTagInfo {
  domain: string
  tagList: string[]
}

@injectable()
export default class TagApis {
  constructor(private client: DomainProxyClient) {}

  getTags() {
    return this.client.get('/domain/all/tags', null, { withProduct: true }).then<string[]>(
      (resp: { tagList: string[] }) => (resp.tagList || [])
    )
  }

  createTag(tag: string) {
    return this.client.post<void>('/domain/tag', { tag }, { withProduct: true })
  }

  updateDomainTags(domain: string, tagList: string[]) {
    return this.client.put<void>(`/domain/${domain}/tags`, { tagList }, { withProduct: true })
  }

  batchGetDomainTags(domains: string[]) {
    return this.client.get('/domain/tags', { domains }).then<DomainTagInfo[]>(
      (resp: { list: DomainTagInfo[] }) => (resp.list || [])
    )
  }
}
