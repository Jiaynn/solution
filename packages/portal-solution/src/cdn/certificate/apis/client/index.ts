export { SslClient } from './ssl'
export { SslProxyClient } from './ssl-proxy'
export { SslHttpClient } from './ssl-http'
