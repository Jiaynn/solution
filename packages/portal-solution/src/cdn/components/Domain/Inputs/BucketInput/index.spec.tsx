/*
 * @file cases for bucket input component
 * @author nighca <nighca@live.cn>
 */

import React from 'react'

import { RendererUtils as Renderer } from 'test'

import mockDomainDetail from 'cdn/test/domain-detail-mock'

import { IBucketSimplified } from 'cdn/apis/bucket'

import DomainBucketInput, { SortType, createState } from '.'
import DomainApis from 'cdn/apis/domain'

const domain = mockDomainDetail()

const buckets: IBucketSimplified[] = [{ name: 'zzzz', zone: 'z0', private: 1, protected: 0, last_operation: '', last_operation_extra: '', last_operation_at: '2018-06-21T10:24:40.965210971+08:00', fop_access_white_list: [], files: 0, storage: 0, line_files: 0, line_storage: 0, right: '', share: false }, { name: 'nao-1', zone: 'na0', private: 0, protected: 0, last_operation: '', last_operation_extra: '', last_operation_at: '2018-06-21T10:24:40.965217486+08:00', fop_access_white_list: [], files: 0, storage: 0, line_files: 0, line_storage: 0, right: '', share: false }, { name: 'hahahha', zone: 'z0', private: 0, protected: 0, last_operation: '', last_operation_extra: '', last_operation_at: '2018-06-21T10:24:40.965222651+08:00', fop_access_white_list: [], files: 0, storage: 0, line_files: 0, line_storage: 0, right: 'RW', share: true }, { name: 'abcd', zone: 'z0', private: 0, protected: 0, last_operation: '', last_operation_extra: '', last_operation_at: '2018-06-21T10:24:40.965223468+08:00', fop_access_white_list: [], files: 0, storage: 0, line_files: 0, line_storage: 0, right: 'RW', share: false }, { name: 'dfaf', zone: 'z1', private: 0, protected: 0, last_operation: '', last_operation_extra: '', last_operation_at: '2018-06-21T10:24:40.965225036+08:00', fop_access_white_list: [], files: 0, storage: 0, line_files: 0, line_storage: 0, right: '', share: false }]

it('renders correctly', () => {
  const renderer = new Renderer()
  const domainApis = renderer.inject(DomainApis)
  const tree = renderer.createWithAct(
    <DomainBucketInput
      domains={[domain]}
      buckets={buckets}
      disabled={false}
      state={createState(null, { domainApis, getDomains: () => ([domain]) })}
    />
  ).toJSON()
  expect(tree).toMatchSnapshot()
})

it('renders correctly with disabled: true', () => {
  const renderer = new Renderer()
  const domainApis = renderer.inject(DomainApis)
  const tree = renderer.createWithAct(
    <DomainBucketInput
      domains={[domain]}
      buckets={buckets}
      disabled
      state={createState(null, { domainApis, getDomains: () => ([domain]) })}
    />
  ).toJSON()
  expect(tree).toMatchSnapshot()
})

it('renders correctly with value', () => {
  const renderer = new Renderer()
  const domainApis = renderer.inject(DomainApis)
  const tree = renderer.createWithAct(
    <DomainBucketInput
      domains={[domain]}
      buckets={buckets}
      disabled={false}
      state={createState(buckets[1].name, { domainApis, getDomains: () => ([domain]), modify: false })}
    />
  ).toJSON()
  expect(tree).toMatchSnapshot()
})

it('renders correctly with invalid value', () => {
  const renderer = new Renderer()
  const domainApis = renderer.inject(DomainApis)
  const tree = renderer.createWithAct(
    <DomainBucketInput
      domains={[domain]}
      buckets={buckets}
      disabled={false}
      state={createState('BUCKET_NOT_EXIST', { domainApis, getDomains: () => ([domain]), modify: false })}
    />
  ).toJSON()
  expect(tree).toMatchSnapshot()
})

it('renders correctly with error', () => {
  const renderer = new Renderer()
  const domainApis = renderer.inject(DomainApis)
  const state = createState('BUCKET_NOT_EXIST', { domainApis, getDomains: () => ([domain]), modify: false })
  state.setError('bucket 不可为空')

  const tree = renderer.createWithAct(
    <DomainBucketInput
      domains={[domain]}
      buckets={buckets}
      disabled={false}
      state={state}
    />
  ).toJSON()
  expect(tree).toMatchSnapshot()
})

it('renders correctly with sort', () => {
  const renderer = new Renderer()
  const domainApis = renderer.inject(DomainApis)
  const tree = renderer.createWithAct(
    <DomainBucketInput
      domains={[domain]}
      buckets={buckets}
      disabled={false}
      sort={SortType.Lexic}
      state={createState(null, { domainApis, getDomains: () => ([domain]) })}
    />
  ).toJSON()
  expect(tree).toMatchSnapshot()
})
