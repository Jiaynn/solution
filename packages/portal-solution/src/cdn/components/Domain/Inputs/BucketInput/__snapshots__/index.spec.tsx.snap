// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="line domain-bucket-input-wrapper"
>
  <div
    className="line"
  >
    <div
      className="text-input-wrapper"
    >
      <div
        className="ant-select ant-select-enabled"
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onMouseDown={[Function]}
        onMouseOut={[Function]}
        onMouseUp={[Function]}
        onTouchStart={[Function]}
      >
        <div
          aria-autocomplete="list"
          aria-controls="test-uuid"
          aria-expanded={false}
          aria-haspopup="true"
          className="ant-select-selection
            ant-select-selection--single"
          onKeyDown={[Function]}
          role="combobox"
          tabIndex={0}
        >
          <div
            className="ant-select-selection__rendered"
          >
            <div
              className="ant-select-selection__placeholder"
              onClick={[Function]}
              onMouseDown={[Function]}
              style={
                Object {
                  "WebkitUserSelect": "none",
                  "display": "none",
                  "userSelect": "none",
                }
              }
              unselectable="on"
            >
              请选择一个 bucket
            </div>
            <div
              className="ant-select-selection-selected-value"
              style={
                Object {
                  "display": "block",
                  "opacity": 1,
                }
              }
              title="abcd"
            >
              abcd
            </div>
            <div
              className="ant-select-search ant-select-search--inline"
              style={
                Object {
                  "display": "none",
                }
              }
            >
              <div
                className="ant-select-search__field__wrap"
              >
                <input
                  autoComplete="off"
                  className="ant-select-search__field"
                  disabled={false}
                  onChange={[Function]}
                  onKeyDown={[Function]}
                  value=""
                />
                <span
                  className="ant-select-search__field__mirror"
                >
                  
                   
                </span>
              </div>
            </div>
          </div>
          <span
            className="ant-select-arrow"
            onClick={[Function]}
            style={
              Object {
                "WebkitUserSelect": "none",
                "userSelect": "none",
              }
            }
            unselectable="on"
          >
            <i
              aria-label="icon: down"
              className="anticon anticon-down ant-select-arrow-icon"
            >
              <svg
                aria-hidden="true"
                className=""
                data-icon="down"
                fill="currentColor"
                focusable="false"
                height="1em"
                viewBox="64 64 896 896"
                width="1em"
              >
                <path
                  d="M884 256h-75c-5.1 0-9.9 2.5-12.9 6.6L512 654.2 227.9 262.6c-3-4.1-7.8-6.6-12.9-6.6h-75c-6.5 0-10.3 7.4-6.5 12.7l352.6 486.1c12.8 17.6 39 17.6 51.7 0l352.6-486.1c3.9-5.3.1-12.7-6.4-12.7z"
                />
              </svg>
            </i>
          </span>
        </div>
      </div>
    </div>
    <div
      className="comp-no-icp-warning"
    >
      <span
        className="icp-text"
      >
        域名未备案，
      </span>
      <span
        className="icp-desc"
      >
        <span>
          只能选择海外存储，您可以点击 
          <a
            className="comp-router-link"
            href="/kodo/bucket?shouldCreateBucket=true"
            onClick={[Function]}
            rel="noopener"
            target="_blank"
          >
            创建海外存储
          </a>
        </span>
      </span>
    </div>
    <span
      className="warning-wrapper extra-tip"
    >
      当源站选择存储时，若发生回源请求将会由存储产生 
      <a
        href="https://www.qiniu.com/prices"
        rel="noopener"
        target="_blank"
      >
        CDN 回源流量费
      </a>
    </span>
  </div>
</div>
`;

exports[`renders correctly with disabled: true 1`] = `
<div
  className="line domain-bucket-input-wrapper"
>
  <div
    className="line"
  >
    <div
      className="text-input-wrapper"
    >
      <div
        className="ant-select ant-select-disabled"
        onBlur={[Function]}
        onFocus={[Function]}
        onMouseDown={[Function]}
        onMouseOut={[Function]}
        onMouseUp={[Function]}
      >
        <div
          aria-autocomplete="list"
          aria-controls="test-uuid"
          aria-expanded={false}
          aria-haspopup="true"
          className="ant-select-selection
            ant-select-selection--single"
          onKeyDown={[Function]}
          role="combobox"
          tabIndex={-1}
        >
          <div
            className="ant-select-selection__rendered"
          >
            <div
              className="ant-select-selection__placeholder"
              onClick={[Function]}
              onMouseDown={[Function]}
              style={
                Object {
                  "WebkitUserSelect": "none",
                  "display": "none",
                  "userSelect": "none",
                }
              }
              unselectable="on"
            >
              请选择一个 bucket
            </div>
            <div
              className="ant-select-selection-selected-value"
              style={
                Object {
                  "display": "block",
                  "opacity": 1,
                }
              }
              title=""
            >
              
            </div>
            <div
              className="ant-select-search ant-select-search--inline"
              style={
                Object {
                  "display": "none",
                }
              }
            >
              <div
                className="ant-select-search__field__wrap"
              >
                <input
                  autoComplete="off"
                  className="ant-select-search__field"
                  disabled={true}
                  onChange={[Function]}
                  onKeyDown={[Function]}
                  value=""
                />
                <span
                  className="ant-select-search__field__mirror"
                >
                  
                   
                </span>
              </div>
            </div>
          </div>
          <span
            className="ant-select-arrow"
            onClick={[Function]}
            style={
              Object {
                "WebkitUserSelect": "none",
                "userSelect": "none",
              }
            }
            unselectable="on"
          >
            <i
              aria-label="icon: down"
              className="anticon anticon-down ant-select-arrow-icon"
            >
              <svg
                aria-hidden="true"
                className=""
                data-icon="down"
                fill="currentColor"
                focusable="false"
                height="1em"
                viewBox="64 64 896 896"
                width="1em"
              >
                <path
                  d="M884 256h-75c-5.1 0-9.9 2.5-12.9 6.6L512 654.2 227.9 262.6c-3-4.1-7.8-6.6-12.9-6.6h-75c-6.5 0-10.3 7.4-6.5 12.7l352.6 486.1c12.8 17.6 39 17.6 51.7 0l352.6-486.1c3.9-5.3.1-12.7-6.4-12.7z"
                />
              </svg>
            </i>
          </span>
        </div>
      </div>
    </div>
    <div
      className="comp-no-icp-warning"
    >
      <span
        className="icp-text"
      >
        域名未备案，
      </span>
      <span
        className="icp-desc"
      >
        <span>
          只能选择海外存储，您可以点击 
          <a
            className="comp-router-link"
            href="/kodo/bucket?shouldCreateBucket=true"
            onClick={[Function]}
            rel="noopener"
            target="_blank"
          >
            创建海外存储
          </a>
        </span>
      </span>
    </div>
    <span
      className="warning-wrapper extra-tip"
    >
      当源站选择存储时，若发生回源请求将会由存储产生 
      <a
        href="https://www.qiniu.com/prices"
        rel="noopener"
        target="_blank"
      >
        CDN 回源流量费
      </a>
    </span>
  </div>
</div>
`;

exports[`renders correctly with error 1`] = `
<div
  className="line domain-bucket-input-wrapper"
>
  <div
    className="line"
  >
    <div
      className="text-input-wrapper"
    >
      <div
        className="ant-select ant-select-enabled"
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onMouseDown={[Function]}
        onMouseOut={[Function]}
        onMouseUp={[Function]}
        onTouchStart={[Function]}
      >
        <div
          aria-autocomplete="list"
          aria-controls="test-uuid"
          aria-expanded={false}
          aria-haspopup="true"
          className="ant-select-selection
            ant-select-selection--single"
          onKeyDown={[Function]}
          role="combobox"
          tabIndex={0}
        >
          <div
            className="ant-select-selection__rendered"
          >
            <div
              className="ant-select-selection__placeholder"
              onClick={[Function]}
              onMouseDown={[Function]}
              style={
                Object {
                  "WebkitUserSelect": "none",
                  "display": "none",
                  "userSelect": "none",
                }
              }
              unselectable="on"
            >
              请选择一个 bucket
            </div>
            <div
              className="ant-select-selection-selected-value"
              style={
                Object {
                  "display": "block",
                  "opacity": 1,
                }
              }
              title="abcd"
            >
              abcd
            </div>
            <div
              className="ant-select-search ant-select-search--inline"
              style={
                Object {
                  "display": "none",
                }
              }
            >
              <div
                className="ant-select-search__field__wrap"
              >
                <input
                  autoComplete="off"
                  className="ant-select-search__field"
                  disabled={false}
                  onChange={[Function]}
                  onKeyDown={[Function]}
                  value=""
                />
                <span
                  className="ant-select-search__field__mirror"
                >
                  
                   
                </span>
              </div>
            </div>
          </div>
          <span
            className="ant-select-arrow"
            onClick={[Function]}
            style={
              Object {
                "WebkitUserSelect": "none",
                "userSelect": "none",
              }
            }
            unselectable="on"
          >
            <i
              aria-label="icon: down"
              className="anticon anticon-down ant-select-arrow-icon"
            >
              <svg
                aria-hidden="true"
                className=""
                data-icon="down"
                fill="currentColor"
                focusable="false"
                height="1em"
                viewBox="64 64 896 896"
                width="1em"
              >
                <path
                  d="M884 256h-75c-5.1 0-9.9 2.5-12.9 6.6L512 654.2 227.9 262.6c-3-4.1-7.8-6.6-12.9-6.6h-75c-6.5 0-10.3 7.4-6.5 12.7l352.6 486.1c12.8 17.6 39 17.6 51.7 0l352.6-486.1c3.9-5.3.1-12.7-6.4-12.7z"
                />
              </svg>
            </i>
          </span>
        </div>
      </div>
    </div>
    <span
      className="error-wrapper"
      style={
        Object {
          "color": "#e55c5c",
          "display": "block",
          "lineHeight": "20px",
          "marginTop": "8px",
          "verticalAlign": "middle",
        }
      }
    >
      bucket 不可为空
    </span>
    <div
      className="comp-no-icp-warning"
    >
      <span
        className="icp-text"
      >
        域名未备案，
      </span>
      <span
        className="icp-desc"
      >
        <span>
          只能选择海外存储，您可以点击 
          <a
            className="comp-router-link"
            href="/kodo/bucket?shouldCreateBucket=true"
            onClick={[Function]}
            rel="noopener"
            target="_blank"
          >
            创建海外存储
          </a>
        </span>
      </span>
    </div>
    <span
      className="warning-wrapper extra-tip"
    >
      当源站选择存储时，若发生回源请求将会由存储产生 
      <a
        href="https://www.qiniu.com/prices"
        rel="noopener"
        target="_blank"
      >
        CDN 回源流量费
      </a>
    </span>
  </div>
</div>
`;

exports[`renders correctly with invalid value 1`] = `
<div
  className="line domain-bucket-input-wrapper"
>
  <div
    className="line"
  >
    <div
      className="text-input-wrapper"
    >
      <div
        className="ant-select ant-select-enabled"
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onMouseDown={[Function]}
        onMouseOut={[Function]}
        onMouseUp={[Function]}
        onTouchStart={[Function]}
      >
        <div
          aria-autocomplete="list"
          aria-controls="test-uuid"
          aria-expanded={false}
          aria-haspopup="true"
          className="ant-select-selection
            ant-select-selection--single"
          onKeyDown={[Function]}
          role="combobox"
          tabIndex={0}
        >
          <div
            className="ant-select-selection__rendered"
          >
            <div
              className="ant-select-selection__placeholder"
              onClick={[Function]}
              onMouseDown={[Function]}
              style={
                Object {
                  "WebkitUserSelect": "none",
                  "display": "none",
                  "userSelect": "none",
                }
              }
              unselectable="on"
            >
              请选择一个 bucket
            </div>
            <div
              className="ant-select-selection-selected-value"
              style={
                Object {
                  "display": "block",
                  "opacity": 1,
                }
              }
              title="abcd"
            >
              abcd
            </div>
            <div
              className="ant-select-search ant-select-search--inline"
              style={
                Object {
                  "display": "none",
                }
              }
            >
              <div
                className="ant-select-search__field__wrap"
              >
                <input
                  autoComplete="off"
                  className="ant-select-search__field"
                  disabled={false}
                  onChange={[Function]}
                  onKeyDown={[Function]}
                  value=""
                />
                <span
                  className="ant-select-search__field__mirror"
                >
                  
                   
                </span>
              </div>
            </div>
          </div>
          <span
            className="ant-select-arrow"
            onClick={[Function]}
            style={
              Object {
                "WebkitUserSelect": "none",
                "userSelect": "none",
              }
            }
            unselectable="on"
          >
            <i
              aria-label="icon: down"
              className="anticon anticon-down ant-select-arrow-icon"
            >
              <svg
                aria-hidden="true"
                className=""
                data-icon="down"
                fill="currentColor"
                focusable="false"
                height="1em"
                viewBox="64 64 896 896"
                width="1em"
              >
                <path
                  d="M884 256h-75c-5.1 0-9.9 2.5-12.9 6.6L512 654.2 227.9 262.6c-3-4.1-7.8-6.6-12.9-6.6h-75c-6.5 0-10.3 7.4-6.5 12.7l352.6 486.1c12.8 17.6 39 17.6 51.7 0l352.6-486.1c3.9-5.3.1-12.7-6.4-12.7z"
                />
              </svg>
            </i>
          </span>
        </div>
      </div>
    </div>
    <div
      className="comp-no-icp-warning"
    >
      <span
        className="icp-text"
      >
        域名未备案，
      </span>
      <span
        className="icp-desc"
      >
        <span>
          只能选择海外存储，您可以点击 
          <a
            className="comp-router-link"
            href="/kodo/bucket?shouldCreateBucket=true"
            onClick={[Function]}
            rel="noopener"
            target="_blank"
          >
            创建海外存储
          </a>
        </span>
      </span>
    </div>
    <span
      className="warning-wrapper extra-tip"
    >
      当源站选择存储时，若发生回源请求将会由存储产生 
      <a
        href="https://www.qiniu.com/prices"
        rel="noopener"
        target="_blank"
      >
        CDN 回源流量费
      </a>
    </span>
  </div>
</div>
`;

exports[`renders correctly with sort 1`] = `
<div
  className="line domain-bucket-input-wrapper"
>
  <div
    className="line"
  >
    <div
      className="text-input-wrapper"
    >
      <div
        className="ant-select ant-select-enabled"
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onMouseDown={[Function]}
        onMouseOut={[Function]}
        onMouseUp={[Function]}
        onTouchStart={[Function]}
      >
        <div
          aria-autocomplete="list"
          aria-controls="test-uuid"
          aria-expanded={false}
          aria-haspopup="true"
          className="ant-select-selection
            ant-select-selection--single"
          onKeyDown={[Function]}
          role="combobox"
          tabIndex={0}
        >
          <div
            className="ant-select-selection__rendered"
          >
            <div
              className="ant-select-selection__placeholder"
              onClick={[Function]}
              onMouseDown={[Function]}
              style={
                Object {
                  "WebkitUserSelect": "none",
                  "display": "none",
                  "userSelect": "none",
                }
              }
              unselectable="on"
            >
              请选择一个 bucket
            </div>
            <div
              className="ant-select-selection-selected-value"
              style={
                Object {
                  "display": "block",
                  "opacity": 1,
                }
              }
              title="abcd"
            >
              abcd
            </div>
            <div
              className="ant-select-search ant-select-search--inline"
              style={
                Object {
                  "display": "none",
                }
              }
            >
              <div
                className="ant-select-search__field__wrap"
              >
                <input
                  autoComplete="off"
                  className="ant-select-search__field"
                  disabled={false}
                  onChange={[Function]}
                  onKeyDown={[Function]}
                  value=""
                />
                <span
                  className="ant-select-search__field__mirror"
                >
                  
                   
                </span>
              </div>
            </div>
          </div>
          <span
            className="ant-select-arrow"
            onClick={[Function]}
            style={
              Object {
                "WebkitUserSelect": "none",
                "userSelect": "none",
              }
            }
            unselectable="on"
          >
            <i
              aria-label="icon: down"
              className="anticon anticon-down ant-select-arrow-icon"
            >
              <svg
                aria-hidden="true"
                className=""
                data-icon="down"
                fill="currentColor"
                focusable="false"
                height="1em"
                viewBox="64 64 896 896"
                width="1em"
              >
                <path
                  d="M884 256h-75c-5.1 0-9.9 2.5-12.9 6.6L512 654.2 227.9 262.6c-3-4.1-7.8-6.6-12.9-6.6h-75c-6.5 0-10.3 7.4-6.5 12.7l352.6 486.1c12.8 17.6 39 17.6 51.7 0l352.6-486.1c3.9-5.3.1-12.7-6.4-12.7z"
                />
              </svg>
            </i>
          </span>
        </div>
      </div>
    </div>
    <div
      className="comp-no-icp-warning"
    >
      <span
        className="icp-text"
      >
        域名未备案，
      </span>
      <span
        className="icp-desc"
      >
        <span>
          只能选择海外存储，您可以点击 
          <a
            className="comp-router-link"
            href="/kodo/bucket?shouldCreateBucket=true"
            onClick={[Function]}
            rel="noopener"
            target="_blank"
          >
            创建海外存储
          </a>
        </span>
      </span>
    </div>
    <span
      className="warning-wrapper extra-tip"
    >
      当源站选择存储时，若发生回源请求将会由存储产生 
      <a
        href="https://www.qiniu.com/prices"
        rel="noopener"
        target="_blank"
      >
        CDN 回源流量费
      </a>
    </span>
  </div>
</div>
`;

exports[`renders correctly with value 1`] = `
<div
  className="line domain-bucket-input-wrapper"
>
  <div
    className="line"
  >
    <div
      className="text-input-wrapper"
    >
      <div
        className="ant-select ant-select-enabled"
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onMouseDown={[Function]}
        onMouseOut={[Function]}
        onMouseUp={[Function]}
        onTouchStart={[Function]}
      >
        <div
          aria-autocomplete="list"
          aria-controls="test-uuid"
          aria-expanded={false}
          aria-haspopup="true"
          className="ant-select-selection
            ant-select-selection--single"
          onKeyDown={[Function]}
          role="combobox"
          tabIndex={0}
        >
          <div
            className="ant-select-selection__rendered"
          >
            <div
              className="ant-select-selection__placeholder"
              onClick={[Function]}
              onMouseDown={[Function]}
              style={
                Object {
                  "WebkitUserSelect": "none",
                  "display": "none",
                  "userSelect": "none",
                }
              }
              unselectable="on"
            >
              请选择一个 bucket
            </div>
            <div
              className="ant-select-selection-selected-value"
              style={
                Object {
                  "display": "block",
                  "opacity": 1,
                }
              }
              title="nao-1"
            >
              nao-1
            </div>
            <div
              className="ant-select-search ant-select-search--inline"
              style={
                Object {
                  "display": "none",
                }
              }
            >
              <div
                className="ant-select-search__field__wrap"
              >
                <input
                  autoComplete="off"
                  className="ant-select-search__field"
                  disabled={false}
                  onChange={[Function]}
                  onKeyDown={[Function]}
                  value=""
                />
                <span
                  className="ant-select-search__field__mirror"
                >
                  
                   
                </span>
              </div>
            </div>
          </div>
          <span
            className="ant-select-arrow"
            onClick={[Function]}
            style={
              Object {
                "WebkitUserSelect": "none",
                "userSelect": "none",
              }
            }
            unselectable="on"
          >
            <i
              aria-label="icon: down"
              className="anticon anticon-down ant-select-arrow-icon"
            >
              <svg
                aria-hidden="true"
                className=""
                data-icon="down"
                fill="currentColor"
                focusable="false"
                height="1em"
                viewBox="64 64 896 896"
                width="1em"
              >
                <path
                  d="M884 256h-75c-5.1 0-9.9 2.5-12.9 6.6L512 654.2 227.9 262.6c-3-4.1-7.8-6.6-12.9-6.6h-75c-6.5 0-10.3 7.4-6.5 12.7l352.6 486.1c12.8 17.6 39 17.6 51.7 0l352.6-486.1c3.9-5.3.1-12.7-6.4-12.7z"
                />
              </svg>
            </i>
          </span>
        </div>
      </div>
    </div>
    <div
      className="comp-no-icp-warning"
    >
      <span
        className="icp-text"
      >
        域名未备案，
      </span>
      <span
        className="icp-desc"
      >
        <span>
          只能选择海外存储，您可以点击 
          <a
            className="comp-router-link"
            href="/kodo/bucket?shouldCreateBucket=true"
            onClick={[Function]}
            rel="noopener"
            target="_blank"
          >
            创建海外存储
          </a>
        </span>
      </span>
    </div>
    <span
      className="warning-wrapper extra-tip"
    >
      当源站选择存储时，若发生回源请求将会由存储产生 
      <a
        href="https://www.qiniu.com/prices"
        rel="noopener"
        target="_blank"
      >
        CDN 回源流量费
      </a>
    </span>
  </div>
</div>
`;
